server/src/index.ts

```diff
@@ -13,82 +14,119 @@ if (!fs.existsSync(uploadDir)) {
+    const content = JSON.parse(buffer.toString());
+    data.push(content);
+  }
   return data;
 };
 
-const deleteEmployee = (fileName: string): boolean => {
-  let deleted = false;
+const deleteEmployee = async (id: string): Promise<string | null> => {
   try {
-    fs.unlinkSync(`${uploadDir}${fileName}`);
-    deleted = true;
+    await unlink(`${uploadDir}${id}.json`);
+    return id;
   } catch (error: unknown) {
     console.error(error);
+    return null;
   }
-  return deleted;
 };
 
-const addEmployee = (fileName: string, content: string): boolean => {
-  let saved = false;
+const addEmployee = async (
+  content: Required<IEmployee>
+): Promise<string | null> => {
+  const { id } = content;
+  const stringify = JSON.stringify(content);
   try {
-    fs.writeFileSync(`${uploadDir}${fileName}`, content);
-    saved = true;
+    await writeFile(`${uploadDir}${id}.json`, stringify);
+    return id;
   } catch (error: unknown) {
     console.error(error);
+    return null;
   }
-  return saved;
 };
 
-const requestListenner = (
+const requestListenner = async (
   req: http.IncomingMessage,
   res: http.ServerResponse
 ) => {
   const { url: endpoint, method } = req;
-  if (method === "POST" && endpoint === ENDPOINTS.addEmployee) {
-    const body: Array<Uint8Array> = [];
-    req
-      .on("error", (error: Error) => console.error(error.message))
-      .on("data", (chunk: Uint8Array) => body.push(chunk))
-      .on("end", () => {
-        let parsedData: EmployeeI = JSON.parse(Buffer.concat(body).toString());
-        const isDataValid = validateEmployeeObject(parsedData);
-        if (isDataValid) {
-          if (!parsedData?.id) parsedData.id = uuid.v1();
-          const json = JSON.stringify(parsedData);
-          const savedSuccessfully = addEmployee(`${parsedData.id}.json`, json);
-          res.end(savedSuccessfully ? json : "Sorry, could not save data.");
+  const withoutParams = endpoint?.split("?")?.[0];
+  switch (withoutParams) {
+    case ENDPOINTS.employees:
+      try {
+        const files = await parseAllFiles();
+        const stringify = JSON.stringify(files);
+        res.end(stringify);
+      } catch (error) {
+        res.writeHead(500);
+        res.end("An error occured when parsing files.");
+      }
+      break;
+    case ENDPOINTS.employee:
+      if (method === "POST") {
+        const body: Array<Uint8Array> = [];
+        req
+          .on("error", (error: Error) => {
+            console.error(error.message);
+          })
+          .on("data", (chunk: Uint8Array) => {
+            body.push(chunk);
+          })
+          .on("end", async () => {
+            const concatBody = Buffer.concat(body).toString();
+            let parsedData: IEmployee = JSON.parse(concatBody);
+            const isDataValid = validateEmployeeObject(parsedData);
+            if (isDataValid) {
+              if (!parsedData?.id) {
+                parsedData.id = uuid.v1();
+              }
+              try {
```

Comment:
It's better to put all the functionality of the controller into try catch

Reply:
Okay, changed